File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* t1=nullptr;
        ListNode* t2=nullptr;
        while(head!=nullptr){
            t1=head->next;
            head->next=t2;
            t2=head;
            head=t1;
        }    
        return t2;
    }
};

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        head=ListNode(10001)
        temp=head
        while len(lists):
            m=head
            c=0
            k=-1
            for i in range(len(lists)):
                if(not lists[i]):
                    c+=1
                    continue
                if lists[i].val<m.val:
                    m=lists[i]
                    k=i
            if c==len(lists):
                break
            lists[k]=m.next
            temp.next=m
            temp=temp.next
        return head.next

                
                
                

            
        